solutions/swb-reference/src/services/sshKeyService.ts

```diff
@@ -222,8 +226,135 @@ export default class SshKeyService implements SshKeyPlugin {
+      throw new Ec2Error(e.message);
+    }
+
+    // send ssh public key
+    try {
+      const { Success: success } = await ec2InstanceConnect.sendSSHPublicKey({
+        InstanceId: instanceId,
+        InstanceOSUser: 'ec2-user',
+        SSHPublicKey: key.PublicKey
+      });
+      if (!success) {
+        throw new Ec2Error(`Could not send SSH Public Key to environment ${environmentId}`);
+      }
+    } catch (e) {
+      throw new Ec2Error(e.message);
+    }
+
+    // return network info
+    const reservations = response.Reservations;
+    if (!reservations || reservations.length !== 1) {
+      throw new NoInstanceFoundError('No instance found');
+    }
+
+    const reservation = reservations[0];
+    return this._getConnectionInfoFromReservation(reservation);
+  }
+
+  /**
+   * Private method to get a single SSH Key from EC2
+   *
+   * @param sshKeyId - the KeyName of the SSH Key to get
+   * @param projectId - the project the key belongs to
+   * @returns {@link KeyPairInfo} with the SSH Key info
+   */
+  private async _getSshKey(sshKeyId: string, projectId: string): Promise<KeyPairInfo> {
+    // get envMgmtRoleArn and externalId from project record
+    const { envMgmtRoleArn, externalId } = await this._getEnvMgmtRoleArnAndExternalIdFromProject(projectId);
+
+    // get EC2 client
+    const { ec2 } = await this._getEc2ClientsForHostingAccount(envMgmtRoleArn, 'Get', externalId, this._aws);
+
+    // get ssh key
+    let keys = [];
+    try {
+      const response = await ec2.describeKeyPairs({
+        Filters: [{ Name: 'key-name', Values: [sshKeyId] }],
+        IncludePublicKey: true
+      });
+      keys = response.KeyPairs || [];
+    } catch (e) {
+      throw new Ec2Error(e);
+    }
+    if (keys.length === 0) {
+      throw new NoKeyExistsError(`Key ${sshKeyId} does not exist`);
+    }
+    if (keys.length > 1) {
+      throw new NonUniqueKeyError(`More than one key exists with ${sshKeyId}.`);
+    }
+    // must only be 1 key
+    return keys[0];
+  }
+
+  /**
+   * Gets the connection info (public DNS name, public IP address, private DNS name, private IP address)
+   * from an EC2 {@link Reservation} for a single instance
+   *
+   * @param reservation - a {@link Reservation} for a single instance
+   * @returns {@link SendPublicKeyResponse} that contains the connection info
+   */
+  private _getConnectionInfoFromReservation(reservation: Reservation): SendPublicKeyResponse {
+    const instances = reservation.Instances;
+    if (!instances || instances.length !== 1) {
+      throw new NoInstanceFoundError('No instance found');
+    }
+
+    const instance = instances[0];
+    const {
+      PublicDnsName: publicDnsName,
+      PublicIpAddress: publicIp,
+      PrivateDnsName: privateDnsName,
+      PrivateIpAddress: privateIp
+    } = instance;
+    if (!(publicDnsName && publicIp && privateDnsName && privateIp)) {
+      throw new ConnectionInfoNotDefinedError('Some connection info is not defined');
```

Comment:
Probably want to specify what is missing to help with debugging.

Reply:
I added this info as well